core/server/common/src/test/java/alluxio/master/journal/raft/RaftJournalTest.java

```diff
@@ -139,6 +139,50 @@ public void suspendCatchupResume() throws Exception {
         () -> countingMaster.getApplyCount() == entryCount + 1, mWaitOptions);
   }
 
+  @Test
+  public void suspendSnapshotRestart() throws Exception {
+    // Create a counting master implementation that counts how many journal entries it processed.
+    CountingDummyFileSystemMaster countingMaster = new CountingDummyFileSystemMaster();
+    mFollowerJournalSystem.createJournal(countingMaster);
+
+    final int entryCount = 10;
+    try (JournalContext journalContext =
+             mLeaderJournalSystem.createJournal(new NoopMaster()).createJournalContext()) {
+      for (int i = 0; i < entryCount; i++) {
+        journalContext.append(
+            alluxio.proto.journal.Journal.JournalEntry.newBuilder().setInodeLastModificationTime(
+                File.InodeLastModificationTimeEntry.newBuilder().setId(i).build()).build());
+      }
+    }
+
+    // Suspend follower journal system.
+    mFollowerJournalSystem.suspend();
+
+    // Write more entries which are not applied due to suspension.
+    try (JournalContext journalContext =
+             mLeaderJournalSystem.createJournal(new NoopMaster()).createJournalContext()) {
+      journalContext
+          .append(alluxio.proto.journal.Journal.JournalEntry.newBuilder()
+              .setInodeLastModificationTime(
+                  File.InodeLastModificationTimeEntry.newBuilder().setId(entryCount).build())
+              .build());
+    }
+
+    // Ask the follower to do a snapshot.
+    mFollowerJournalSystem.checkpoint();
+
+    // Restart the follower.
+    mFollowerJournalSystem.stop();
```

Comment:
why added restart follower here

Reply:
Restart causes the follower snapshot to be replayed so its content can be verified.